vhd.c

```diff
@@ -0,0 +1,87 @@
+// vhd.c: Check for a Virtual Hard Disk file
+#define _GNU_SOURCE
+#include "config.h"
+#include <endian.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include "dat.h"
+#include "fns.h"
+
+struct _DiskGeometry
+{
+	uint16_t cylinder;
+	uint8_t heads;
+	uint8_t sectors;
+} __attribute__ ((packed));
+typedef struct _DiskGeometry DiskGeometry;
+
+struct _HardDiskFooter
+{
+    char cookie[8];
+    uint32_t features;
+    uint32_t fileFormatVersion;
+    uint64_t dataOffset;
+    uint32_t timeStamp;
+    uint32_t creatorApplication;
+    uint32_t creatorVersion;
+    uint32_t creatorHostOS;
+    uint64_t originalSize;
+    uint64_t currentSize;
+    DiskGeometry diskGeometry;
+    uint32_t diskType;
+    uint32_t checksum;
+    uint64_t uniqueId[2];
+    uint8_t savedState;
+    uint8_t reserved[427];
+} __attribute__ ((packed));
```

Comment:
If this was copied from somewhere else, can you please tell me the source?  I want to make sure the license is compatible.  Or if it's from documentation, a comment to that effect (with a reference to the specific documentation) would be nice to have near here.